src/codegen/translate-expression.lisp

```diff
@@ -934,6 +934,15 @@ Returns a `node'.")
        :type (tc:qualified-ty-type qual-ty)
        :name (tc:pattern-var-name pat))))
 
+  (:method ((pat tc:pattern-binding))
+    (let ((qual-ty (tc:pattern-type pat)))
+      (assert (null (tc:qualified-ty-predicates qual-ty)))
```

Comment:
is this a coalton-bug compiler assertion, or an assertion against the user?

Reply:
not sure - very much copy-and-paste to look like all the other methods. I'll try to understand why

Reply:
Most of the methods in `translate-expression` expect the `qualified-ty` instance in their type field to actually be unqualified. Not all of them, but most. The only genuine exception is in the translation of `node-var`, where `apply-dicts` is called, where the node unqualified concrete type is extracted and applied to the returned node. 

If we're going to yank out a concrete type from a `qualified-ty`, we want to make sure that there aren't any dangling predicates that we've are yet to contend with.